### DISCUSSION TITLE
Common project depending on `Uno.WinUI` breaks Windows

### DISCUSSION THREAD

**Discussion by marwalsch** [unoplatform/uno#7965] [General]
I have a few .NET Standard projects on which the Shared is supposed to depend on. Since the latter can't do so by itself I included the dependency inside each platform's head as with every other dependency. 

However, one of these projects depends on `Uno.WinUI` (the `Windows.Storage` API specifically) and thus breaks the Desktop head.

What is the recommended approach to deal with this? Managing everything inside the single shared project would work but obviously a poor choice for several reasons.

**Comment by jeromelaban** [comment]
Please see [this document](https://platform.uno/docs/articles/uno-app-solution-structure.html?q=shared%20project#understanding-shared-projects) for a shared projects explainer.

**Comment by marwalsch** [comment]
For Googlers:

After some tampering I figured that, in theory, the `dotnet new unolib` template would be the ideal option (e.g. for component vendors). I ended up doing the following however:

- Moving everything not directly UI-related into class libraries.
- Resolve unavoidable dependencies like the one mentioned above through IoC (depend on an Interface inside the class-library that the main app has to provide an implementation for (not beautiful but it works).
